### PR TITLE
py-python-socks: Submission

### PR DESCRIPTION
**Pull request by essandess**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-curio/Portfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+checksums           rmd160  4f422c83ccfa3b13a771eb745bb3f5d61fa67201 \
+                    sha256  af08212e590bb7da8e4cc39c42012711494dc20d622f162155ba296cc2e3bc10 \
+                    size    234920
+
+python.versions     38 39 310
+
+if {${name} ne ${subport}} {
+    depends_run-append  \
+                    port:py${python.version}-setuptools
```
Comment (reneeotten) [inline python/py-curio/Portfile:28]: why is this a runtime dependency? I see it listed in `setup.py` in a `try` clause, it appears you decided not to use it as a build dependency; for what is it used at runtime?

File: python/py-python-socks/Portfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-python-socks
+version             2.0.3
+revision            0
+license             Apache2
```
Comment (reneeotten) [inline python/py-python-socks/Portfile:9]: this should be `Apache-2` as `port lint` tells you

File: python/py-python-socks/Portfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    depends_run-append  \
+                    port:py${python.version}-anyio \
+                    port:py${python.version}-async-timeout \
+                    port:py${python.version}-attrs \
+                    port:py${python.version}-curio \
+                    port:py${python.version}-flask \
+                    port:py${python.version}-psutil \
+                    port:py${python.version}-trio \
+                    port:py${python.version}-yarl
+}
```
Comment (reneeotten) [inline python/py-python-socks/Portfile:41]: there are only four optional dependencies listed in the [README](https://github.com/romis2012/python-socks/tree/v2.0.3#requirements) and in [`setup.py`](https://github.com/romis2012/python-socks/blob/c43f061905f1d7384127efce4708c8da26ef63cc/setup.py#L54) - where do all the others come from?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-curio/Portfile (hunk 1 above)
Comment (essandess) [reply python/py-curio/Portfile:28]: Oops. Fixed thanks.

File: python/py-python-socks/Portfile (hunk 2 above)
Comment (essandess) [reply python/py-python-socks/Portfile]: Fixed thanks.

File: python/py-python-socks/Portfile (hunk 3 above)
Comment (essandess) [reply python/py-python-socks/Portfile:39]: Fixed thanks after double-checking with searches, e.g. https://github.com/romis2012/python-socks/search?q=yarl.

Comment (reneeotten) [reply python/py-python-socks/Portfile:39]: I'm not sure whether I agree that this is fixed. If you look that the search you mention in your comment above you'll notice that it is only imported in tests. So why is it runtime dependency and not a test dependency? Again, only the four dependencies that are listed as optional as I stated in my first comment, are used in the actual code. All others seem to be only used in tests and I would assume are, therefore, only test dependencies - or?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-python-socks/Portfile (hunk 3 above)
Comment (essandess) [reply python/py-python-socks/Portfile:36]: Okay now I think I have it right. Thank you very much for pointing me in the right direction.

**Review by reneeotten — APPROVED** [review]

**Pull request merged**